packages/mui-joy/src/ToggleButtonGroup/ToggleButtonGroup.test.tsx

```diff
@@ -44,6 +45,80 @@ describe('<ToggleButtonGroup />', () => {
     expect(secondButton).to.have.property('disabled', true);
   });
 
+  it('should toggle the state when immediate children is not button', () => {
+    function CustomButton() {
+      return (
+        <div>
+          <Button value="custom-button">Custom button</Button>
+        </div>
+      );
+    }
+    function CustomIconButton() {
+      return (
+        <div>
+          <IconButton value="custom-iconbutton">Custom button</IconButton>
+        </div>
+      );
+    }
+    const handleChange = spy();
+
+    function ToggleGroup() {
+      const [value, setValue] = React.useState([]);
+      return (
+        <ToggleButtonGroup
+          onChange={(...args) => {
+            setValue(args[1]!);
+            handleChange(...args);
+          }}
+          value={value}
+        >
+          <Tooltip title="button 1">
+            <Button value="tooltip-button">one</Button>
+          </Tooltip>
+          <CustomButton />
+          <CustomIconButton />
+          <Tooltip title="button 2">
+            <IconButton value="tooltip-iconbutton">two</IconButton>
+          </Tooltip>
+        </ToggleButtonGroup>
+      );
+    }
+
+    const { getAllByRole } = render(<ToggleGroup />);
+
+    const [firstButton, secondButton, thirdButton, fouthButton] = getAllByRole('button');
+    act(() => {
```

Comment:
Why do we need `act` here?

Reply:
if not for using `act` i'm seeing below error from `rtl`.


```
Warning: An update to ToggleGroup inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):
```

Reply:
Oh okay, because we are changing state and the component re-renders.